### PR TITLE
Model accuracy alignment

### PR DESCRIPTION
**Pull request by zhengshengning**
### PR Category

Operator Mechanism

### PR Types

New features

### Description

1. rotary_pos_emb_output 层输出无法对齐修改：self.inv_freq 原来调用GPU算子，现在改为CPU算子
2. 使用 Eager Attention 替代 Flash Attention
3. Fleet自实现Linear Diff
4. attention_mask 从 flaot32 类型 cast bool 

### 是否引起精度变化

否

### CODE REVIEW

**Review by wanghuancoder — DISMISSED** [review]
LGTM

File: src/paddlefleet/models/gpt/gpt_embedding.py [hunk 1]
```diff
@@ -112,6 +112,9 @@ def __init__(
                 rotary_interleaved=config.rotary_interleaved,
                 rotary_base=rotary_base,
                 rope_scaling=rope_scaling,
+                use_accuracy_compatible=getattr(
```
Comment (From00) [inline src/paddlefleet/models/gpt/gpt_embedding.py:115]: 这个开关是否可以统一通过yaml进行配置？

File: src/paddlefleet/transformer/moe/fp8_utils.py [hunk 2]
```diff
@@ -940,9 +942,14 @@ def bwd_down_input_bf16(self, expert_w2, unzipped_grad, o1, unzipped_probs):
                         start_idx:end_idx
                     ].contiguous()
                     expert_w2_i = expert_w2[i].T.contiguous()
-                    do2_s_list.append(
-                        F.linear(x=unzipped_grad_i, weight=expert_w2_i)
-                    )
+                    if self.use_accuracy_compatible:
+                        do2_s_list.append(
```
Comment (From00) [inline src/paddlefleet/transformer/moe/fp8_utils.py:946]: 这里需要考虑带bias的情况吗？

File: src/paddlefleet/transformer/dot_product_attention.py [hunk 3]
```diff
@@ -609,6 +645,35 @@ def forward(
...
+                    self.scale_mask_softmax.input_in_float16 = True
+                else:
+                    self.scale_mask_softmax.input_in_fp16 = False
+                    self.scale_mask_softmax.input_in_bf16 = False
+                    self.scale_mask_softmax.input_in_float16 = False
+
+        # PaddleFormers collate emits float32 lower-triangle masks where 1.0 means attend and 0.0
+        # means mask. PaddleFleet mask_func expects bool masks where True means
+        # masked-out, so convert to strict upper-triangle semantics.
+        if (
```
Comment (From00) [inline src/paddlefleet/transformer/dot_product_attention.py:670]: 这个是为了跟PaddleFormers对齐吗？

File: src/paddlefleet/models/gpt/gpt_embedding.py (hunk 1 above)
Comment (zhengshengning) [reply src/paddlefleet/models/gpt/gpt_embedding.py:115]: 可以，已在PaddleFormers增加了支持：https://github.com/PaddlePaddle/PaddleFormers/pull/4692

File: src/paddlefleet/transformer/moe/fp8_utils.py (hunk 2 above)
Comment (zhengshengning) [reply src/paddlefleet/transformer/moe/fp8_utils.py:946]: 目前精度对齐的几个模型中都没有bias的情况

File: src/paddlefleet/transformer/dot_product_attention.py (hunk 3 above)
Comment (zhengshengning) [reply src/paddlefleet/transformer/dot_product_attention.py:670]: 这个是为了和Megatron对齐，在minimax模型对齐时，Megatron同一位置用的attention_mask时bool类型，我们的是float，所以这里cast一下。不知道其他模型是不是同一情况，后面可以验证一下

**Review by From00 — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]

**Review by sneaxiy — APPROVED** [review]
